tests/ignite/handlers/test_checkpoint.py

```diff
@@ -96,95 +99,96 @@ def test_checkpoint_score_function_wrong_output():
         checkpointer(trainer)
 
 
-def test_checkpoint_default():
-    def _test(to_save, obj, name):
-        save_handler = MagicMock(spec=BaseSaveHandler)
-
-        checkpointer = Checkpoint(to_save, save_handler=save_handler)
-        assert checkpointer.last_checkpoint is None
-
-        trainer = Engine(lambda e, b: None)
-        trainer.state = State(epoch=0, iteration=0)
-
-        checkpointer(trainer)
-        assert save_handler.call_count == 1
+@pytest.mark.parametrize(
```

Comment:
Following the same idea of my previous remark, what do you think about someting like :
```python
def test_matrix():
    model = DummyModel()
    optimizer = torch.optim.SGD(model.parameters(), lr=0.1)

    return [
        ({"model": model}, model.state_dict(), "model"),
        (
            {"model": model, "optimizer": optimizer},
            {"model": model.state_dict(), "optimizer": optimizer.state_dict()},
            "checkpoint",
        ),
    ]

@pytest.mark.parametrize("to_save, obj, name", test_matrix())
test_checkpoint_default(to_save, obj, name):
   ...
```

Reply:
Edit, we can forget this.